File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
  const clothes = Array(n).fill(1);

  lost.map((item) => (clothes[item - 1] = 0));

  reserve.map((item) => clothes[item - 1]++);

  for (let i = 0; i < n; i++) {
    // 체육복이 0개인 학생이 앞사람한테 받아왔을 때
    if (clothes[i] === 0 && clothes[i - 1] === 2) {
      clothes[i] = 1;
      clothes[i - 1] = 1;
    }
    // 체육복이 0개인 학생이 뒷사람한테 받아왔을 때
    else if (clothes[i] === 0 && clothes[i + 1] === 2) {
      clothes[i] = 1;
      clothes[i + 1] = 1;
    }
  }

  return clothes.filter((item) => item > 0).length;
}

File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    students = [1 for _ in range(n)]
    
    for i in lost:
        students[i - 1] -= 1
    
    for i in reserve:
        students[i - 1] += 1
        
    for i in range(n):
        if students[i] == 0:
            # 왼쪽 확인
            if 0 <= i - 1 and students[i - 1] == 2:
                students[i] = 1
                students[i - 1] -= 1
            # 오른쪽 확인    
            elif i + 1 < n and students[i + 1] == 2:
                students[i] = 1
                students[i + 1] -= 1
                
    return n - students.count(0)